### ISSUE TITLE
Binding <C-n> and <C-p> to switch buffers not working in normal mode

### ISSUE THREAD

**Issue by fghimis** [GermainZ/weechat-vimode#92]
I am trying to reuse the normal keybindings for switching channels. 

Normally, when pressing Ctrl+n I will advance one channel and Ctrl+p will go back.
Currently, this works only one I'm in Insert mode. When I switch to Normal mode it stops working.

I have tried the following nmaps:
 - :nmap  \<C-n\> /buffer +1\<CR\>
 - :nmap  \<C-N\> /buffer +1\<CR\>
 - :nmap  ^N /buffer +1\<CR\>
 - :nmap  C-N /buffer +1\<CR\>

And none worked. 
I'm using the _latest_ script on https://weechat.org/scripts/source/vimode.py.html/ :  Version: 0.8.1

More info (I don't know if it helps):
- on pressing `:nmap` after setting the keybindings, I would not get the expected \<C-n\>, but only n or N or ^N.
- also ^F and ^B don't seem to work with default settings, but ^U and ^D work fine

**Comment by GermainZ** [comment]
Please try the script from this repo, that should work.
(The versions are a bit messed up, 0.8/0.8.1 from the weechat repo are actually 0.7 in this repo).

I will submit a PR to update the weechat script when I have some time :)

**Comment by fghimis** [comment]
Thanks, it works now as intended!

**Comment by fghimis** [comment]
Another question, not to open another issue, but when I'm in normal mode and I press <Esc> I would get back 1 character (like backspace). Can that be fixed?

**Comment by GermainZ** [comment]
That's intended behavior (similar to vim). You can edit the source code if you don't like it, though. Specifically, you can comment out/remove [this line](https://github.com/GermainZ/weechat-vimode/blob/897a33b9fb28c98c4e0a1c292d13536dd57f85c7/vimode.py#L2111).

**Comment by fghimis** [comment]
No, I mean when I'm in Insert mode and i press Esc it's normal to go back 1 character.
But I am in **Normal mode**, and I press Esc _again_, and I go back 1 character (similar to pressing h in Normal mode).
Shouldn't it do nothing (already being in Normal mode)?

**Comment by GermainZ** [comment]
Oh yeah, I'll reopen the issue and rename it so I don't forget about it. Thanks!

**Comment by fghimis** [comment]
Oh and something else (it may be related to mode switching): Pressing <Esc> to cancel Ex mode will erase current line. Steps to reproduce:
1) Enter insert mode, write: "some text here"
2) Press Esc (enter normal mode)
3) Press : (enter ex mode)
4) Change your mind and press Esc to go back in normal mode
5) "some text here" is gone, only : remains
